docs/guide.md

```diff
@@ -195,24 +195,31 @@ provider set:
 
 ```go
 type Fooer interface {
-    Foo() string
+    Foo() Baz
```

Comment:
I think this example would be clearer with less distinct type names. Let's keep `Foo()` returning a `string`....

Reply:
Done.